// viem@2.30.6/_esm/errors/transaction.js
import { BaseError as t } from "./base.js";
function i(a) {
  const e = Object.entries(a).map(([n, r]) => r === void 0 || r === !1 ? null : [n, r]).filter(Boolean), o = e.reduce((n, [r]) => Math.max(n, r.length), 0);
  return e.map(([n, r]) => `  ${`${n}:`.padEnd(o + 1)}  ${r}`).join(`
`);
}
class c extends t {
  constructor() {
    super([
      "Cannot specify both a `gasPrice` and a `maxFeePerGas`/`maxPriorityFeePerGas`.",
      "Use `maxFeePerGas`/`maxPriorityFeePerGas` for EIP-1559 compatible networks, and `gasPrice` for others."
    ].join(`
`), { name: "FeeConflictError" });
  }
}
class d extends t {
  constructor({ v: e }) {
    super(`Invalid \`v\` value "${e}". Expected 27 or 28.`, {
      name: "InvalidLegacyVError"
    });
  }
}
class l extends t {
  constructor({ transaction: e }) {
    super("Cannot infer a transaction type from provided transaction.", {
      metaMessages: [
        "Provided Transaction:",
        "{",
        i(e),
        "}",
        "",
        "To infer the type, either provide:",
        "- a `type` to the Transaction, or",
        "- an EIP-1559 Transaction with `maxFeePerGas`, or",
        "- an EIP-2930 Transaction with `gasPrice` & `accessList`, or",
        "- an EIP-4844 Transaction with `blobs`, `blobVersionedHashes`, `sidecars`, or",
        "- an EIP-7702 Transaction with `authorizationList`, or",
        "- a Legacy Transaction with `gasPrice`"
      ],
      name: "InvalidSerializableTransactionError"
    });
  }
}
class p extends t {
  constructor({ storageKey: e }) {
    super(`Size for storage key "${e}" is invalid. Expected 32 bytes. Got ${Math.floor((e.length - 2) / 2)} bytes.`, { name: "InvalidStorageKeySizeError" });
  }
}
export {
  c as FeeConflictError,
  d as InvalidLegacyVError,
  l as InvalidSerializableTransactionError,
  p as InvalidStorageKeySizeError,
  i as prettyPrint
};
