// @0xsequence/indexer@2.3.17/dist/0xsequence-indexer.esm.js
function d() {
  return d = Object.assign ? Object.assign.bind() : function(u) {
    for (var r = 1; r < arguments.length; r++) {
      var o = arguments[r];
      for (var e in o) ({}).hasOwnProperty.call(o, e) && (u[e] = o[e]);
    }
    return u;
  }, d.apply(null, arguments);
}
let ee = /* @__PURE__ */ function(u) {
  return u.VERIFIED = "VERIFIED", u.UNVERIFIED = "UNVERIFIED", u.ALL = "ALL", u;
}({});
const X = "Webrpc", Y = "webrpc@v0.21.1;gen-typescript@v0.15.1;sequence-indexer@v0.4.0";
class Z {
  constructor(r, o) {
    this.hostname = void 0, this.fetch = void 0, this.path = "/rpc/IndexerGateway/", this.getNativeTokenBalance = (e, s, t) => this.fetch(this.url("GetNativeTokenBalance"), i(e, s, t)).then((n) => h(n).then((a) => ({
      balances: a.balances
    })), (n) => {
      throw p.new({
        cause: `fetch(): ${n.message || ""}`
      });
    }), this.getTokenBalances = (e, s, t) => this.fetch(this.url("GetTokenBalances"), i(e, s, t)).then((n) => h(n).then((a) => ({
      page: a.page,
      balances: a.balances
    })), (n) => {
      throw p.new({
        cause: `fetch(): ${n.message || ""}`
      });
    }), this.getTokenBalancesSummary = (e, s, t) => this.fetch(this.url("GetTokenBalancesSummary"), i(e, s, t)).then((n) => h(n).then((a) => ({
      page: a.page,
      nativeBalances: a.nativeBalances,
      balances: a.balances
    })), (n) => {
      throw p.new({
        cause: `fetch(): ${n.message || ""}`
      });
    }), this.getTokenBalancesDetails = (e, s, t) => this.fetch(this.url("GetTokenBalancesDetails"), i(e, s, t)).then((n) => h(n).then((a) => ({
      page: a.page,
      nativeBalances: a.nativeBalances,
      balances: a.balances
    })), (n) => {
      throw p.new({
        cause: `fetch(): ${n.message || ""}`
      });
    }), this.getTokenBalancesByContract = (e, s, t) => this.fetch(this.url("GetTokenBalancesByContract"), i(e, s, t)).then((n) => h(n).then((a) => ({
      page: a.page,
      balances: a.balances
    })), (n) => {
      throw p.new({
        cause: `fetch(): ${n.message || ""}`
      });
    }), this.getBalanceUpdates = (e, s, t) => this.fetch(this.url("GetBalanceUpdates"), i(e, s, t)).then((n) => h(n).then((a) => ({
      page: a.page,
      balances: a.balances
    })), (n) => {
      throw p.new({
        cause: `fetch(): ${n.message || ""}`
      });
    }), this.getChains = (e, s, t) => this.fetch(this.url("GetChains"), i(e, s, t)).then((n) => h(n).then((a) => ({
      chains: a.chains
    })), (n) => {
      throw p.new({
        cause: `fetch(): ${n.message || ""}`
      });
    }), this.ping = (e, s) => this.fetch(this.url("Ping"), i({}, e, s)).then((t) => h(t).then((n) => ({
      status: n.status
    })), (t) => {
      throw p.new({
        cause: `fetch(): ${t.message || ""}`
      });
    }), this.version = (e, s) => this.fetch(this.url("Version"), i({}, e, s)).then((t) => h(t).then((n) => ({
      version: n.version
    })), (t) => {
      throw p.new({
        cause: `fetch(): ${t.message || ""}`
      });
    }), this.runtimeStatus = (e, s) => this.fetch(this.url("RuntimeStatus"), i({}, e, s)).then((t) => h(t).then((n) => ({
      status: n.status
    })), (t) => {
      throw p.new({
        cause: `fetch(): ${t.message || ""}`
      });
    }), this.hostname = r.replace(/\/*$/, ""), this.fetch = (e, s) => o(e, s);
  }
  url(r) {
    return this.hostname + this.path + r;
  }
}
const i = (u = {}, r = {}, o = null) => {
  const e = d({}, r, {
    "Content-Type": "application/json"
  });
  return e[X] = Y, {
    method: "POST",
    headers: e,
    body: JSON.stringify(u || {}),
    signal: o
  };
}, h = (u) => u.text().then((r) => {
  let o;
  try {
    o = JSON.parse(r);
  } catch (e) {
    let s = "";
    throw e instanceof Error && (s = e.message), l.new({
      status: u.status,
      cause: `JSON.parse(): ${s}: response text: ${r}`
    });
  }
  if (!u.ok) {
    const e = typeof o.code == "number" ? o.code : 0;
    throw (_[e] || c).new(o);
  }
  return o;
});
class c extends Error {
  constructor(r, o, e, s, t) {
    super(e), this.name = void 0, this.code = void 0, this.message = void 0, this.status = void 0, this.cause = void 0, this.msg = void 0, this.name = r || "WebrpcError", this.code = typeof o == "number" ? o : 0, this.message = e || `endpoint error ${this.code}`, this.msg = this.message, this.status = typeof s == "number" ? s : 0, this.cause = t, Object.setPrototypeOf(this, c.prototype);
  }
  static new(r) {
    return new this(r.error, r.code, r.message || r.msg, r.status, r.cause);
  }
}
class f extends c {
  constructor(r = "WebrpcEndpoint", o = 0, e = "endpoint error", s = 0, t) {
    super(r, o, e, s, t), Object.setPrototypeOf(this, f.prototype);
  }
}
class p extends c {
  constructor(r = "WebrpcRequestFailed", o = -1, e = "request failed", s = 0, t) {
    super(r, o, e, s, t), Object.setPrototypeOf(this, p.prototype);
  }
}
class b extends c {
  constructor(r = "WebrpcBadRoute", o = -2, e = "bad route", s = 0, t) {
    super(r, o, e, s, t), Object.setPrototypeOf(this, b.prototype);
  }
}
class O extends c {
  constructor(r = "WebrpcBadMethod", o = -3, e = "bad method", s = 0, t) {
    super(r, o, e, s, t), Object.setPrototypeOf(this, O.prototype);
  }
}
class x extends c {
  constructor(r = "WebrpcBadRequest", o = -4, e = "bad request", s = 0, t) {
    super(r, o, e, s, t), Object.setPrototypeOf(this, x.prototype);
  }
}
class l extends c {
  constructor(r = "WebrpcBadResponse", o = -5, e = "bad response", s = 0, t) {
    super(r, o, e, s, t), Object.setPrototypeOf(this, l.prototype);
  }
}
class g extends c {
  constructor(r = "WebrpcServerPanic", o = -6, e = "server panic", s = 0, t) {
    super(r, o, e, s, t), Object.setPrototypeOf(this, g.prototype);
  }
}
class m extends c {
  constructor(r = "WebrpcInternalError", o = -7, e = "internal error", s = 0, t) {
    super(r, o, e, s, t), Object.setPrototypeOf(this, m.prototype);
  }
}
class j extends c {
  constructor(r = "WebrpcClientDisconnected", o = -8, e = "client disconnected", s = 0, t) {
    super(r, o, e, s, t), Object.setPrototypeOf(this, j.prototype);
  }
}
class P extends c {
  constructor(r = "WebrpcStreamLost", o = -9, e = "stream lost", s = 0, t) {
    super(r, o, e, s, t), Object.setPrototypeOf(this, P.prototype);
  }
}
class w extends c {
  constructor(r = "WebrpcStreamFinished", o = -10, e = "stream finished", s = 0, t) {
    super(r, o, e, s, t), Object.setPrototypeOf(this, w.prototype);
  }
}
class v extends c {
  constructor(r = "Unauthorized", o = 1e3, e = "Unauthorized access", s = 0, t) {
    super(r, o, e, s, t), Object.setPrototypeOf(this, v.prototype);
  }
}
class B extends c {
  constructor(r = "PermissionDenied", o = 1001, e = "Permission denied", s = 0, t) {
    super(r, o, e, s, t), Object.setPrototypeOf(this, B.prototype);
  }
}
class A extends c {
  constructor(r = "SessionExpired", o = 1002, e = "Session expired", s = 0, t) {
    super(r, o, e, s, t), Object.setPrototypeOf(this, A.prototype);
  }
}
class R extends c {
  constructor(r = "MethodNotFound", o = 1003, e = "Method not found", s = 0, t) {
    super(r, o, e, s, t), Object.setPrototypeOf(this, R.prototype);
  }
}
class E extends c {
  constructor(r = "RequestConflict", o = 1004, e = "Conflict with target resource", s = 0, t) {
    super(r, o, e, s, t), Object.setPrototypeOf(this, E.prototype);
  }
}
class k extends c {
  constructor(r = "Aborted", o = 1005, e = "Request aborted", s = 0, t) {
    super(r, o, e, s, t), Object.setPrototypeOf(this, k.prototype);
  }
}
class I extends c {
  constructor(r = "Geoblocked", o = 1006, e = "Geoblocked region", s = 0, t) {
    super(r, o, e, s, t), Object.setPrototypeOf(this, I.prototype);
  }
}
class W extends c {
  constructor(r = "RateLimited", o = 1007, e = "Rate-limited. Please slow down.", s = 0, t) {
    super(r, o, e, s, t), Object.setPrototypeOf(this, W.prototype);
  }
}
class $ extends c {
  constructor(r = "ProjectNotFound", o = 1100, e = "Project not found", s = 0, t) {
    super(r, o, e, s, t), Object.setPrototypeOf(this, $.prototype);
  }
}
class S extends c {
  constructor(r = "AccessKeyNotFound", o = 1101, e = "Access key not found", s = 0, t) {
    super(r, o, e, s, t), Object.setPrototypeOf(this, S.prototype);
  }
}
class T extends c {
  constructor(r = "AccessKeyMismatch", o = 1102, e = "Access key mismatch", s = 0, t) {
    super(r, o, e, s, t), Object.setPrototypeOf(this, T.prototype);
  }
}
class N extends c {
  constructor(r = "InvalidOrigin", o = 1103, e = "Invalid origin for Access Key", s = 0, t) {
    super(r, o, e, s, t), Object.setPrototypeOf(this, N.prototype);
  }
}
class F extends c {
  constructor(r = "InvalidService", o = 1104, e = "Service not enabled for Access key", s = 0, t) {
    super(r, o, e, s, t), Object.setPrototypeOf(this, F.prototype);
  }
}
class G extends c {
  constructor(r = "UnauthorizedUser", o = 1105, e = "Unauthorized user", s = 0, t) {
    super(r, o, e, s, t), Object.setPrototypeOf(this, G.prototype);
  }
}
class q extends c {
  constructor(r = "QuotaExceeded", o = 1200, e = "Quota exceeded", s = 0, t) {
    super(r, o, e, s, t), Object.setPrototypeOf(this, q.prototype);
  }
}
class K extends c {
  constructor(r = "RateLimit", o = 1201, e = "Rate limit exceeded", s = 0, t) {
    super(r, o, e, s, t), Object.setPrototypeOf(this, K.prototype);
  }
}
class U extends c {
  constructor(r = "NoDefaultKey", o = 1300, e = "No default access key found", s = 0, t) {
    super(r, o, e, s, t), Object.setPrototypeOf(this, U.prototype);
  }
}
class C extends c {
  constructor(r = "MaxAccessKeys", o = 1301, e = "Access keys limit reached", s = 0, t) {
    super(r, o, e, s, t), Object.setPrototypeOf(this, C.prototype);
  }
}
class D extends c {
  constructor(r = "AtLeastOneKey", o = 1302, e = "You need at least one Access Key", s = 0, t) {
    super(r, o, e, s, t), Object.setPrototypeOf(this, D.prototype);
  }
}
class L extends c {
  constructor(r = "Timeout", o = 1900, e = "Request timed out", s = 0, t) {
    super(r, o, e, s, t), Object.setPrototypeOf(this, L.prototype);
  }
}
class M extends c {
  constructor(r = "InvalidArgument", o = 2001, e = "Invalid argument", s = 0, t) {
    super(r, o, e, s, t), Object.setPrototypeOf(this, M.prototype);
  }
}
class z extends c {
  constructor(r = "Unavailable", o = 2002, e = "Unavailable resource", s = 0, t) {
    super(r, o, e, s, t), Object.setPrototypeOf(this, z.prototype);
  }
}
class V extends c {
  constructor(r = "QueryFailed", o = 2003, e = "Query failed", s = 0, t) {
    super(r, o, e, s, t), Object.setPrototypeOf(this, V.prototype);
  }
}
class H extends c {
  constructor(r = "ResourceExhausted", o = 2004, e = "Resource exhausted", s = 0, t) {
    super(r, o, e, s, t), Object.setPrototypeOf(this, H.prototype);
  }
}
class Q extends c {
  constructor(r = "NotFound", o = 3e3, e = "Resource not found", s = 0, t) {
    super(r, o, e, s, t), Object.setPrototypeOf(this, Q.prototype);
  }
}
class J extends c {
  constructor(r = "MetadataCallFailed", o = 3003, e = "Metadata service call failed", s = 0, t) {
    super(r, o, e, s, t), Object.setPrototypeOf(this, J.prototype);
  }
}
const _ = {
  0: f,
  [-1]: p,
  [-2]: b,
  [-3]: O,
  [-4]: x,
  [-5]: l,
  [-6]: g,
  [-7]: m,
  [-8]: j,
  [-9]: P,
  [-10]: w,
  1e3: v,
  1001: B,
  1002: A,
  1003: R,
  1004: E,
  1005: k,
  1006: I,
  1007: W,
  1100: $,
  1101: S,
  1102: T,
  1103: N,
  1104: F,
  1105: G,
  1200: q,
  1201: K,
  1300: U,
  1301: C,
  1302: D,
  1900: L,
  2001: M,
  2002: z,
  2003: V,
  2004: H,
  3e3: Q,
  3003: J
};
class te extends Z {
  constructor(r, o, e) {
    super(r.endsWith("/") ? r.slice(0, -1) : r, fetch), this.projectAccessKey = o, this.jwtAuth = e, this._fetch = (s, t) => {
      const n = {}, a = this.jwtAuth, y = this.projectAccessKey;
      return a && a.length > 0 && (n.Authorization = `BEARER ${a}`), y && y.length > 0 && (n["X-Access-Key"] = y), t.headers = d({}, t.headers, n), fetch(s, t);
    }, this.fetch = this._fetch;
  }
}
export {
  ee as ContractVerificationStatus,
  te as SequenceIndexerGateway
};
