// @noble/hashes@1.8.0/esm/sha2.js
import { HashMD as D, SHA256_IV as b, Chi as g, Maj as p } from "./_md.js";
import { createHasher as u, clean as C, rotr as i } from "./utils.js";
const B = /* @__PURE__ */ Uint32Array.from([
  1116352408,
  1899447441,
  3049323471,
  3921009573,
  961987163,
  1508970993,
  2453635748,
  2870763221,
  3624381080,
  310598401,
  607225278,
  1426881987,
  1925078388,
  2162078206,
  2614888103,
  3248222580,
  3835390401,
  4022224774,
  264347078,
  604807628,
  770255983,
  1249150122,
  1555081692,
  1996064986,
  2554220882,
  2821834349,
  2952996808,
  3210313671,
  3336571891,
  3584528711,
  113926993,
  338241895,
  666307205,
  773529912,
  1294757372,
  1396182291,
  1695183700,
  1986661051,
  2177026350,
  2456956037,
  2730485921,
  2820302411,
  3259730800,
  3345764771,
  3516065817,
  3600352804,
  4094571909,
  275423344,
  430227734,
  506948616,
  659060556,
  883997877,
  958139571,
  1322822218,
  1537002063,
  1747873779,
  1955562222,
  2024104815,
  2227730452,
  2361852424,
  2428436474,
  2756734187,
  3204031479,
  3329325298
]), h = /* @__PURE__ */ new Uint32Array(64);
class E extends D {
  constructor(o = 32) {
    super(64, o, 8, !1), this.A = b[0] | 0, this.B = b[1] | 0, this.C = b[2] | 0, this.D = b[3] | 0, this.E = b[4] | 0, this.F = b[5] | 0, this.G = b[6] | 0, this.H = b[7] | 0;
  }
  get() {
    const { A: o, B: r, C: t, D: s, E: a, F: f, G: c, H: e } = this;
    return [o, r, t, s, a, f, c, e];
  }
  // prettier-ignore
  set(o, r, t, s, a, f, c, e) {
    this.A = o | 0, this.B = r | 0, this.C = t | 0, this.D = s | 0, this.E = a | 0, this.F = f | 0, this.G = c | 0, this.H = e | 0;
  }
  process(o, r) {
    for (let x = 0; x < 16; x++, r += 4)
      h[x] = o.getUint32(r, !1);
    for (let x = 16; x < 64; x++) {
      const H = h[x - 15], d = h[x - 2], m = i(H, 7) ^ i(H, 18) ^ H >>> 3, l = i(d, 17) ^ i(d, 19) ^ d >>> 10;
      h[x] = l + h[x - 7] + m + h[x - 16] | 0;
    }
    let { A: t, B: s, C: a, D: f, E: c, F: e, G: n, H: A } = this;
    for (let x = 0; x < 64; x++) {
      const H = i(c, 6) ^ i(c, 11) ^ i(c, 25), d = A + H + g(c, e, n) + B[x] + h[x] | 0, l = (i(t, 2) ^ i(t, 13) ^ i(t, 22)) + p(t, s, a) | 0;
      A = n, n = e, e = c, c = f + d | 0, f = a, a = s, s = t, t = d + l | 0;
    }
    t = t + this.A | 0, s = s + this.B | 0, a = a + this.C | 0, f = f + this.D | 0, c = c + this.E | 0, e = e + this.F | 0, n = n + this.G | 0, A = A + this.H | 0, this.set(t, s, a, f, c, e, n, A);
  }
  roundClean() {
    C(h);
  }
  destroy() {
    this.set(0, 0, 0, 0, 0, 0, 0, 0), C(this.buffer);
  }
}
const y = /* @__PURE__ */ u(() => new E());
export {
  E as SHA256,
  y as sha256
};
